src/libs/ReportUtils.ts

```diff
@@ -23,171 +26,193 @@ import isReportMessageAttachment from './isReportMessageAttachment';
 import * as defaultWorkspaceAvatars from '../components/Icon/WorkspaceDefaultAvatars';
 import * as CurrencyUtils from './CurrencyUtils';
 import * as UserUtils from './UserUtils';
+import {Beta, Login, PersonalDetails, Policy, Report, ReportAction, Transaction} from '../types/onyx';
+import {Receipt} from '../types/onyx/Transaction';
+import DeepValueOf from '../types/utils/DeepValueOf';
+import {Closed, IOUMessage} from '../types/onyx/OriginalMessage';
+import {Message, ReportActions} from '../types/onyx/ReportAction';
+
+type WelcomeMessage = {showReportName: boolean; phrase1?: string; phrase2?: string};
+
+type Avatar = {
+    id: number;
+    source: React.FC<SvgProps> | string | undefined;
+    type: typeof CONST.ICON_TYPE_AVATAR | typeof CONST.ICON_TYPE_WORKSPACE;
+    name: string;
+    fallbackIcon?: React.FC<SvgProps> | string;
+};
+
+type ExpanseOriginalMessage = {
+    oldComment?: string;
+    newComment?: string;
+    comment?: string;
+    merchant?: string;
+    oldCreated?: string;
+    created?: string;
+    oldMerchant?: string;
+    oldAmount?: number;
+    amount?: number;
+    oldCurrency?: string;
+    currency?: string;
+    category?: string;
+    oldCategory?: string;
+    tag?: string;
+    oldTag?: string;
+    billable?: string;
+    oldBillable?: string;
+};
 
-let currentUserEmail;
-let currentUserAccountID;
-let isAnonymousUser;
+type Participant = {
+    accountID: number;
+    alternateText: string;
+    firstName: string;
+    icons: Avatar[];
+    keyForList: string;
+    lastName: string;
+    login: string;
+    phoneNumber: string;
+    searchText: string;
+    selected: boolean;
+    text: string;
+};
+
+function isTypeTransaction(arg: Transaction | Record<string, never>): arg is Transaction {
+    return arg !== undefined; // Customize this type guard as needed
+}
+
+function isTypeReportAction(arg: ReportAction | Record<string, never>): arg is ReportAction {
+    return arg !== undefined; // Customize this type guard as needed
+}
+
+let currentUserEmail: string | undefined;
+let currentUserAccountID: number | undefined;
```

Comment:
@kubabutkiewicz I've tried to make `email` and `accountId` fields required in the `Session` type, it didn't give errors in ts, and this way, you can get rid of `undefined` here and extra checks moving forward. Let me know if it makes sense for you.